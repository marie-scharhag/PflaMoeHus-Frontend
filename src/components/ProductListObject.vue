<template>
  <ul class="productlistline">
    <a id="hitbox" @click="openproduct()" >
      <div class="productListObject">
        <img
          v-bind:src="'/api/product/picture/'+product.allPictures[0].id"
          alt="Picture"
          class="pic"
        />
        <div class="bg">
          <li>{{ product.name }}</li>
          <li>{{ product.price }} â‚¬</li>
        </div>
      </div>
    </a>
  </ul>
</template>

<script lang="ts">
import "../service/Product";
import { defineComponent } from "vue";

export default defineComponent({
  name: "ProductListObject",
  props: {
    product: Object,
  },
  setup(props, context) {
    /**
     * sends opening event to CompProduct
     */
    function openproduct(): void {
    
      context.emit("open-prod", props.product);
    }

    return {
      openproduct,
    };
  },
});
</script>

<style scoped lang="scss">
.pic {
  border-radius: 10px 10px 0px 0px;
  width: 300px;
  height: 300px;
  object-fit: cover;
  object-position: bottom center;
}
.bg {
  border-radius: 0px 0px 10px 10px;
  padding-top: 0.7em;
  padding-bottom: 0.7em;
  background-color: $color-grey;
  width: 300px;
  text-align: center;
}
#hitbox{
  cursor: pointer;
}

ul {
  list-style-type: none;
}
</style>