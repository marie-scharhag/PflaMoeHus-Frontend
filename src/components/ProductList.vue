<template>
    <div id="productList">
        <ul>
                <ProductListObject id="listOrder" :product="pr" v-for="pr in productlist" :key="pr.id"/> 
        </ul>
        
    </div>
</template>

<script lang="ts">
import ProductListObject from "../components/ProductListObject.vue"
import { useProduct } from "../service/ProductStore";
import { computed, onMounted } from 'vue';

export default{
    components:{
        ProductListObject
    },
    setup(){

        const{allproductslist,update} = useProduct();

        /**
         * calls update function on initialization of this component
         */
        onMounted(async () => {
            await update();
        });
        /**
         * current list of products
         */
        const productlist = computed(() => {
        
          return allproductslist.value;
        
        });
        const imgage = computed

        return{ productlist}; 
    } 
}
</script>

<style scoped lang="scss">
    #listOrder{
        display: inline;
        float: left;
        width: 27.5%;
        margin-bottom: 20px;

    } 

    #productList{
        float: left;
        max-width: 83%;
        margin-left: 15%;
    } 
</style>