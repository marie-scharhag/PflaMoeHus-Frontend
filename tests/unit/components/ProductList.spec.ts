import { mount } from "@vue/test-utils"
	
import ProductList from "@/components/ProductList.vue"
import ProductListObject from "@/components/ProductList.vue"
	
describe('ProductList.vue Test ', () => {
	
    const cmp = mount(ProductList)
	
    test('has the expected html structur', () => {
        expect(cmp.html()).toMatchSnapshot()
    })
	
    test('test if fakeproductlistobject ist used', () => {

        const ProductListObjectMock = {	
            name: 'ProductListObject',
            template: 'fakeProductListObject'
        }
        const wrapper = mount(ProductList, {
            global: {
            stubs: { productListObject: ProductListObjectMock }
            }
        }) 
        expect(wrapper.find('fakeProductListObject')).toBeTruthy() 
    })
})