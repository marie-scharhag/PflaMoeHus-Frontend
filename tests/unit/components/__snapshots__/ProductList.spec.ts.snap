// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList.vue Test  has the expected html structur 1`] = `
<div id="productList">
  <ul></ul>
</div>
`;
